infra/gravity/provision.go

```diff
@@ -144,31 +113,36 @@ func Provision(ctx context.Context, t *testing.T, baseConfig *ProvisionerConfig)
 		}(node)
 	}
 
-	gravityNodes := make([]Gravity, 0, len(nodes))
+	gravityNodes := []Gravity{}
 	errors := []error{}
 	for range nodes {
 		select {
 		case <-ctx.Done():
 			return nil, nil, trace.Errorf("timed out waiting for nodes to provision")
 		case res := <-resultCh:
-			if err != nil {
+			if res.err != nil {
 				errors = append(errors, res.err)
 			} else {
 				gravityNodes = append(gravityNodes, res.gravity)
 			}
 		}
 	}
 
+	logFn := utils.Logf(t, baseConfig.Tag())
 	if len(errors) != 0 {
-		return nil, p.Destroy, trace.NewAggregate(errors...)
+		aggError := trace.NewAggregate(errors...)
+		logFn("cleanup after error provisioning : %v", aggError)
+
+		destroyFn(ctx, t)
```

Comment:
If the `destroyFn` is called here, what happens to it once it gets returned as well?